src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,464 @@
+                        ilk_,
+                        pool,
+                        Art - lineWad,
+                        availableAssets,
+                        Mode.NORMAL,
+                        currentAssets
+                    );
+                } else {
+                    uint256 amount = targetAssets - currentAssets;
+                    if (_add(Art, amount) > lineWad) { // we do not have enough room in the debt ceiling to fully wind
+                        amount = lineWad - Art;
+                    }
+                    _wind(ilk_, pool, amount);
+                }
+            } else if (targetAssets < currentAssets) {
+                _unwind(
+                    ilk_,
+                    pool,
+                    currentAssets - targetAssets,
+                    availableAssets,
+                    Mode.NORMAL,
+                    currentAssets
+                );
+            }
+        }
+    }
+
+    // --- Collect Interest ---
+    function reap(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(live == 1, "DssDirectDepositHub/no-reap-during-cage");
+        require(ilks[ilk_].tic == 0, "DssDirectDepositHub/pool-not-live");
+
+        pool.accrueIfNeeded();
+        uint256 assetBalance = pool.assetBalance();
+        (, uint256 daiDebt) = vat.urns(ilk_, address(pool));
+        if (assetBalance > daiDebt) {
+            uint256 fees = assetBalance - daiDebt;
+            uint256 availableAssets = pool.maxWithdraw();
+            if (fees > availableAssets) {
+                fees = availableAssets;
+            }
+            pool.withdraw(fees);
+            daiJoin.join(vow, fees);
+            emit Reap(ilk_, fees);
+        }
+    }
+
+    // --- Allow DAI holders to exit during global settlement ---
+    // wad: should be amount of gems, this could be different than the share tokens
+    function exit(bytes32 ilk_, address usr, uint256 wad) external {
+        require(wad <= 2 ** 255, "DssDirectDepositHub/overflow");
+        vat.slip(ilk_, msg.sender, -int256(wad));
+        D3MPoolLike pool = ilks[ilk_].pool;
+        require(pool.transfer(usr, wad), "DssDirectDepositHub/failed-transfer");
+        emit Exit(ilk_, usr, wad);
+    }
+
+    // --- Shutdown ---
+    function cage(bytes32 ilk_) external {
+        require(vat.live() == 1, "DssDirectDepositHub/no-cage-during-shutdown");
+
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        // Can shut pools down if we are authed
+        // or if the interest rate strategy changes
+        // or if the main module is caged
+        require(
+            wards[msg.sender] == 1 ||
+            live == 0 ||
+            !pool.validTarget()
+        , "DssDirectDepositHub/not-authorized");
+
+        pool.cage();
+        ilks[ilk_].tic = block.timestamp;
+        emit Cage(ilk_);
+    }
+
+    function cage() external auth {
```

Comment:
I think @hexonaut pointed this out, but what is the reason to have a general `cage`? I think it would be just easier governance cages each one if needed.

Reply:
https://github.com/makerdao/dss-direct-deposit/pull/31/commits/fb0d05c4af31daf8fa919a967567a891748904a2